src/optional.h

```diff
@@ -22,10 +22,19 @@ class Optional {
   T* operator->() {
     return &this->value;
   }
-  
+  void operator=(const T& value) {
+    this->value = value;
+    this->has = true;
+  }
+
   bool has_value() const {
     return has;
   }
+
+  void reset() {
+    this->value = T();
```

Comment:
This seems a bit suspect. I don't think it'll work for non-class objects. I'm actually surprised that the `Optional<tct_thrd_t>` in timer_posix.h compiles, since `tct_thrd_t` is typedef'ed as `pthread_t` (on Posix).

Reply:
In Chromium, they apparently implement `Optional` using a union: https://chromium.googlesource.com/chromium/src/+/master/docs/optional.md#How-is-it-implemented